portality/bll/services/urlshort.py

```diff
@@ -0,0 +1,84 @@
+import random
+import string
+import threading
+from typing import Optional
+
+from portality import models
+from portality.core import app
+from portality.models.url_shortener import AliasQuery, UrlQuery
+from portality.util import url_for
+
+# global current status of the alias length
+alias_len = 6
+ALIAS_CHARS = string.ascii_letters + string.digits
+alias_lock = threading.Lock()
```

Comment:
I am not a fan of these globals; each service worker will have their own one and I'm not sue threading locking will affect that, I think they're separate processes (I may be wrong).

I'd suggest just not doing this, and instead start with a minimum alias size and increase it per request if needed.  The number of aliases we're going to generate is small, collisions are extremely unlikely, and even if it gets very full the number of retries will never be large.

Set the minimum alias length in `settings.py` and then we can increase it if we really need to later.

Reply:
It will become no logic to defend duplicate alias issues If remove both lock and alias count checking.

CPU have no real Random logic, if the program or computer restarted, the program choice the same seed to generate "random" number, the collision will be happen and those bugs is very difficult to be debug.


I will
* remove lock
* remove auto increase aliases length
* keep alias exit checking